### DISCUSSION TITLE
How to query the endpoint_cpm of different instances under the same service

### DISCUSSION THREAD

**Discussion by SmileLnight** [apache/skywalking#10156] [Q&A] [answered]
Hi Team,
OAP Version : 8.5.0
skywalking-ui : 8.5.0
my query：
```
{
    "query": "\nquery queryData($condition: TopNCondition!, $duration: Duration!) {\n  sortMetrics: sortMetrics(condition: $condition, duration: $duration) {\n    name\n    id\n    value\n    refId\n  }}",
    "variables": {
        "duration": {
            "end": "2022-12-13 1157",
            "start": "2022-12-13 1156",
            "step": "MINUTE"
        },
        "condition": {
            "name": "endpoint_cpm",
            "scope": "Endpoint",
            "order": "DES",
            "parentService": "shiyu-svc",
            "normal": true,
            "topN": 500
        }
    }
}
```
It can query the ranking of all endpoint_cpm under the shiyu-svc service, but there are multiple instances of the shiyu-svc service, how can I query the ranking of all endpoint_cpm under a single instance?
Thanks in Advance!

**Comment by wu-sheng** [comment, marked as answer]
This was discussed multiple times over the years. You can't afford this at the storage level, that is the reality.
One service could have 100 instances, and 10k+ endpoints, if we run endpoint per instance statistics, you face million-level metrics per minute for a single service. Your storage would crash without a doubt.

**Reply by SmileLnight** [reply]
Ok, thank you for your reply. What you mean is that the current skywalking only has the service level for the statistical dimension of all endpoint_cpm rankings under a service. If you want to achieve the instance level, the storage will bear additional exponential pressure. Can I understand this?

**Reply by wu-sheng** [reply]
Maybe not real `exponential`, but it would be very crazy.

**Reply by SmileLnight** [reply]
ok thank you very much
